sdk/python/kfp_tekton/compiler/_data_passing_rewriter.py

```diff
@@ -412,6 +422,21 @@ def mark_upstream_ios_of_output(template_output, marked_inputs,
                 or task['taskRef']['name'] in resource_template_names
             ]
 
+            # tekton results doesn't support underscore
+            for argument in task['params']:
+                argument_value = argument.get('value')
+                argument_placeholder_parts = deconstruct_tekton_single_placeholder(
+                    argument_value)
+                if argument_placeholder_parts \
+                    and argument_placeholder_parts[0] == 'tasks' \
+                    and argument_placeholder_parts[1] \
+                    and argument_placeholder_parts[2] \
+                        and argument_placeholder_parts[3]:
```

Comment:
I meant, replace that:

```Python
                if argument_placeholder_parts \
                    and argument_placeholder_parts[0] == 'tasks' \
                    and argument_placeholder_parts[1] \
                    and argument_placeholder_parts[2] \
                        and argument_placeholder_parts[3]:
```

with this:
```Python
                if len(argument_placeholder_parts) == 4 \
                        and argument_placeholder_parts[0] == 'tasks':
```

And change the method `deconstruct_` from:

```Python
def deconstruct_tekton_single_placeholder(s: str) -> List[str]:
    if not re.fullmatch('\\$\\([-._a-zA-Z0-9]+\\)', s):
        return None
    return s.lstrip('$(').rstrip(')').split('.')
```

to:
```Python
def deconstruct_tekton_single_placeholder(s: str) -> List[str]:
    if not re.fullmatch(r'\$\([-._a-zA-Z0-9]+\)', s):  # use an R-string
        return []                                      # return empty list instead of None
    return s.lstrip('$(').rstrip(')').split('.')
```

Reply:
OK, got your mean, thanks.